src/tests/Logary.Facade.Tests/Facade.fs

```diff
@@ -62,33 +79,125 @@ let tests =
       let logger = Targets.create level
       Message.event Verbose "Hi {name}" |> logger.logSimple
 
-    testCase "extract matches" <| fun _ ->
-      let str = "Hi {ho}, we're { something going on  } special"
+    testCase "extract invalid property tokens as text" <| fun _ ->
+      let template = "Hi {ho:##.###}, we're { something going on } special"
       let fields = Map [ "ho", box "hola"; "something going on", box "very" ]
-      let subject = Formatting.extractMatches fields str
-      Expect.equal subject.Length 2 "Should match two things"
-      Expect.equal (subject |> List.map (fun (name, _, _) -> name))
-                   ["ho"; "something going on"]
-                   "Should extract correct names"
-      Expect.equal (subject |> List.map (fun (_, _, m) -> m.Value))
-                   ["{ho}"; "{ something going on  }"]
-                   "Should extract correct match values"
-
-    testCase "extract matches evil" <| fun _ ->
-      let strEvil = "{a}{b}{  c}{d  } {e} fghij {k}} {l} m {{}}"
-      let aThroughL =['a'..'l'] |> List.map string 
-      let fields = aThroughL |> List.map (fun s -> s, s) |> Map.ofList
+      let tokens = parseTemplateTokens template
+      Expect.equal tokens.Length 5 "Extracts 4 texts parts and 1 property part"
+      Expect.equal tokens
+                   [
+                    TextToken ("Hi ")
+                    PropToken ("ho", "##.###")
+                    TextToken (", we're ") // <- a quirk of the parser
+                    TextToken ("{ something going on }")
+                    TextToken (" special")
+                   ]
+                   "Should extract correct name and format parts"
+
+    testCase "handles evil property strings correctly" <| fun _ ->
+      let template = "{a}{b}{  c}{d  } {e} fghij {k}} {l} m {{}}"
+      // validity:     +  +  --    --   +         +    +     --
+      let expectedPropertyNames = ["a"; "b"; "e"; "k"; "l"]
```

Comment:
Because space isn't valid in a property name, is there some good reason to allow it? Logary won't allow it either, neither does: aspnet, serilog, nlog, MessageTemplates, string.format, etc